[Toolchain] Run compile using default toolchain

jyoungyun opened the pull request:
This commit runs compile with cfg using default toolchain.

ONE-vscode-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

YongseopKim reviewed (approved):
LGTM

llFreetimell commented on src/Toolchain/ToolchainEnv.ts:
```diff
@@ -113,11 +114,20 @@ class ToolchainEnv extends Env {
+                           .reduce((r, a) => {
+                             return r.concat(a);
+                           });
+    if (toolchains.length > 0) {
+      // TODO(jyoung) What item will be default toolchain?
+      this.toolchain = toolchains[0];
+    }
```
I think "setting default toolchain" is not related with "getting installed toolchain".
It would be better to separate this later :)

jyoungyun replied:
I think new installed toolchain will be the default toolchain. Because user installed this toolchain to use it. 

And there is no way to set the default toolchain as of now. So this is the only way to use this toolchain when user request to compile.

llFreetimell commented on src/Toolchain/ToolchainEnv.ts:
```diff
@@ -170,14 +183,20 @@ class ToolchainEnv extends Env {
       const job = new JobUninstall(toolchain.uninstall());
-      job.successCallback = () => resolve(true);
+      job.successCallback = () => {
+        // TODO(jyoung) What toolchain will be default item?
```
Is this comment related with uninstalling...? @.@

YongseopKim replied:
AFAIU,

>>> After uninstalling, which toolchain should be as default?

jyoungyun replied:
If the default toolchain is uninstalled, the default toolchain should be changed to another one.

llFreetimell replied:
Ah... I thought that `undefined` is proper to be set after uninstalling.
If other default toolchain should be set after uninstalling, this comment is valid :)

hyunsik-yoon commented on src/OneExplorer/OneExplorer.ts:
```diff
@@ -516,79 +515,3 @@ export class OneExplorer {
-    if (val.exitCode !== undefined && val.exitCode === 0) {
-      vscode.window.showInformationMessage(`Successfully completed.`);
-    } else if (val.intentionallyKilled !== undefined && val.intentionallyKilled === true) {
-      vscode.window.showInformationMessage(`The job was cancelled.`);
```
'Cancel' seems not supported in new change. 
I wonder if 'Cancel' would be added later in different PR.

jyoungyun replied:
I will make another PR for this. This changes will be applied all Job runner events.

jyoungyun replied:
@hyunsik-yoon I made a PR for this comment. #979 Please see this PR.

hyunsik-yoon commented:
Running this PR failed in my test setting. 😢 

This test setting works well with code in main branch

<details>

log

```
[6/27/2022, 1:48:11 PM][ToolchainProvider][info] Compile toolchain: triv2-toolchain-latest-1.1.0~22062620
[6/27/2022, 1:48:11 PM][Env][info] Done building WorkFlow on Env: [{"jobType":9,"name":"config","valid":true,"tool":"onecc","toolArgs":["--config","/home/eric/src/model/ONE-vscode-res/inception_v3/2020-02-03/inception_v3_conv2d.0to128.cfg"],"root":false}]
[6/27/2022, 1:48:11 PM][Helpers][info] obtainWorkspaceRoot: /home/eric/src/model/ONE-vscode-res
[6/27/2022, 1:48:11 PM][JobRunner][info] Run tool: onecc args: ["--config","/home/eric/src/model/ONE-vscode-res/inception_v3/2020-02-03/inception_v3_conv2d.0to128.cfg"] cwd: /home/eric/src/model/ONE-vscode-res root: false
[6/27/2022, 1:48:11 PM][ToolRunner][info] Running: config
tflite2circle: ERROR: Failed to open file: ./inception_v3_conv2d.0to128.tflite
[6/27/2022, 1:48:12 PM][ToolRunner][info] child process exited with code 255
[6/27/2022, 1:48:12 PM][ToolRunner][info] Build Failed: 255
```

my setting:

```
$ pushd /home/eric/src/model/ONE-vscode-res/inception_v3/2020-02-03/
~/src/model/ONE-vscode-res/inception_v3/2020-02-03 ~/src/model/ONE-vscode-res

eric@labs:~/src/model/ONE-vscode-res/inception_v3/2020-02-03  ( ) (13:49:53)
$ ll
total 350212
drwxr-xr-x 2 eric eric     4096  5월 25 17:21 ./
drwxr-xr-x 4 eric eric     4096  5월  6 12:16 ../
-rw-r--r-- 1 eric eric      552  6월 27 13:45 inception_v3_conv2d.0to128.cfg
-rw-r--r-- 1 eric eric 95307124  5월  3 19:29 inception_v3_conv2d.0to128.tflite
-rw-r--r-- 1 eric eric      617  6월 24 14:42 inception_v3_conv2d.0to128-wrongcfg.cfg
-rw-rw-r-- 1 eric eric 95307140  6월 27 13:45 model.circle
-rw-rw-r-- 1 eric eric       84  6월 27 13:45 model.circle.log
-rw-rw-r-- 1 eric eric 95303824  6월 27 13:45 model.opt.circle
-rw-rw-r-- 1 eric eric       96  6월 27 13:45 model.opt.circle.log
-rw-rw-r-- 1 eric eric 24358864  6월 27 13:45 model.q8.circle
-rw-rw-r-- 1 eric eric      642  6월 27 13:45 model.q8.circle.log

eric@labs:~/src/model/ONE-vscode-res/inception_v3/2020-02-03  ( ) (13:49:56)
$ more inception_v3_conv2d.0to128.cfg
[onecc]
one-import-tf=False
one-import-tflite=True
one-import-bcq=False
one-import-onnx=False
one-optimize=True
one-quantize=True
one-pack=False
one-codegen=True

[one-import-tflite]
input_path=./inception_v3_conv2d.0to128.tflite
output_path=./model.circle

[one-optimize]
input_path=./model.circle
output_path=./model.opt.circle
remove_redundant_transpose=True

[one-quantize]
input_path=./model.opt.circle
output_path=./model.q8.circle
granularity=channel

[one-codegen]
backend=tv2
command=-o model.tvn --DSP-quota 8K --save-temps ./model.q8.circle
```




</details>

hyunsik-yoon commented on src/Toolchain/ToolchainProvider.ts:
```diff
@@ -161,7 +166,25 @@ export class ToolchainProvider implements vscode.TreeDataProvider<ToolchainNode>
+        .run(cfg, activeToolchain)
+        .then(
+            () => {
+              vscode.window.showInformationMessage('Compile successfully');
```
How about (since 'Compile successfully` sounds like imperative sentence)?

```suggestion
              vscode.window.showInformationMessage('Successfully Compiled.');
```

llFreetimell reviewed (dismissed):
LGTM

llFreetimell reviewed (dismissed):
LGTM

hyunsik-yoon commented:
@jyoungyun 
When I try this PR, there was an error.

My company system currently blocks me from uploading gif file again. So I'd like to leave a log.
Here you can find password input window is acting weird.

<details>

pw dialog shows from 4:28:09 and 4:28:18

```
[6/30/2022, 4:28:09 PM][Env][info] Done building WorkFlow on Env: [{"jobType":11,"name":"install","valid":true,"tool":"aptitude","toolArgs":["install","-o Aptitude::ProblemResolver::SolutionCost='100*canceled-actions,200*removals'","triv2-toolchain=1.1.0","-q","-y"],"root":true}]
[6/30/2022, 4:28:09 PM][Helpers][info] obtainWorkspaceRoot: /home/eric/m/src/model/ONE-vscode-res
[6/30/2022, 4:28:09 PM][Env][info] Showing password prompt
[6/30/2022, 4:28:09 PM][JobRunner][info] Run tool: aptitude args: ["install","-o Aptitude::ProblemResolver::SolutionCost='100*canceled-actions,200*removals'","triv2-toolchain=1.1.0","-q","-y"] workdir: /home/eric/m/src/model/ONE-vscode-res root: true
[6/30/2022, 4:28:09 PM][ToolRunner][info] Running: install
[sudo] password for eric: Sorry, try again.
[sudo] password for eric: 
sudo: 1 incorrect password attempt
[6/30/2022, 4:28:11 PM][ToolRunner][info] child process exited with code 1
[6/30/2022, 4:28:11 PM][ToolRunner][info] Build Failed: 1

[6/30/2022, 4:28:11 PM][ToolchainProvider][err] Installation is failed
[6/30/2022, 4:28:18 PM][Env][info] Password dialog canceled

```

</details>

jyoungyun commented:
> When I try this PR, there was an error. I removed `out`, `node_modules` and tried to run once more but I got the same result. 😿 

@hyunsik-yoon I fixed this issue. Could you please test it again?

jyoungyun commented:
<img src="https://user-images.githubusercontent.com/7223627/177960624-31b43618-2c14-4f9e-a5ff-3a02df028eb2.gif" width=700 />

/cc @hyunsik-yoon

jyoungyun commented:
Merged. Done!
